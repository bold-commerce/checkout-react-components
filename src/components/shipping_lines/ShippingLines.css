.FieldSet--ShippingMethod .RadioButton {
  border: 1px solid #D9D9D9;
  margin-bottom: 0;
  height: 60px;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 20px;
  cursor: pointer;
  position: relative;
}

.FieldSet--ShippingMethod .RadioButton .stx-radio-field__label {
  position: absolute;
    left: 0;
    height: 100%;
    top: 0;
    padding-left: 45px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    cursor: pointer;
    width: 100%;
}

.FieldSet--ShippingMethod .RadioButton .stx-field {
  position: unset;
}
.FieldSet--ShippingMethod .RadioButton .stx-radio {
  position: unset;
}

.FieldSet--ShippingMethod .RadioButton .stx-radio-field {
  display: flex;
  align-items: center;
}

.FieldSet--ShippingMethod .RadioField {
  margin-bottom: 0;
  padding-left: 0;
}

.FieldSet--ShippingMethod .ShippingMethod__Price .stx-price__amount {
  font-size: 14px;
  line-height: 20px;
  font-weight: normal;
}
