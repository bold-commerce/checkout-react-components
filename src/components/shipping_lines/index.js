// eslint-disable-next-line import/prefer-default-export
export { default as ShippingLines } from './ShippingLines';
