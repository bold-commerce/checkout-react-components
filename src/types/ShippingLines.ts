export interface ShippingLines {
  id: string,
  amount: number,
  description: string
};