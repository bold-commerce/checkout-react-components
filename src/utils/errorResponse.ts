export default interface ErrorResponse {
  type: string,
  payload: any
  error: Error
}