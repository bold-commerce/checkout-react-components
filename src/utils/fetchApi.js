import FetchError from './fetchError';

const fetchApi = async (url, options) => {
  try {
    const response = await fetch(url, {
      mode: 'cors',
      method: 'GET',
      credentials: 'include',
      ...options,
    });

    const responseData = await response.json();

    if (response.status === 200) {
      return {
        success: true,
        data: responseData.data,
      };
    }

    if (response.status === 422) {
      return {
        success: false,
        error: new FetchError(response.status, 'Invalid request', responseData),
      };
    }

    return {
      success: false,
      error: new FetchError(response.status, 'Something went wrong'),
    };
  } catch (e) {
    return {
      success: false,
      error: new FetchError(500, 'Something went wrong'),
    };
  }
};

export default fetchApi;
