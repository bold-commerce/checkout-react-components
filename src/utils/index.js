// eslint-disable-next-line import/prefer-default-export
export { default as fetchApi } from './fetchApi';
export { default as FetchError } from './fetchError';
