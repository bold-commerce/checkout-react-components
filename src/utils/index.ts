// eslint-disable-next-line import/prefer-default-export
export { default as fetchApi } from './fetchApi';
export { default as FetchError } from './fetchError';
export { default as PromiseError } from './promiseError';
export { default as OrderError } from './orderError';
export { default as handleError } from './handleError';
