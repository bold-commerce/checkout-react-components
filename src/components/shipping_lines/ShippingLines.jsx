/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import RadioField from '@boldcommerce/stacks-ui/lib/components/radiofield/RadioField';
import { Price } from '@boldcommerce/stacks-ui';
import EmptyState from '../empty_state/EmptyState';
import LoadingState from '../loading_state/LoadingState';
import { useShippingLines } from '../../hooks';
import './ShippingLines.css';

export const ShippingLines = ({
  showShippingLines, shippingLinesFetching, shippingLinesLoadingStatus, shippingLines, selectedShippingLineIndex, setSelectedShippingLine,
}) => {
  if (!showShippingLines) {
    return (
      <section className="FieldSet FieldSet--ShippingMethod">
        <div className="FieldSet__Header">
          <div className="FieldSet__Heading">Shipping method</div>
        </div>
        <div className="FieldSet__Content"><EmptyState title="To view shipping options, complete filling in your address" /></div>
      </section>
    );
  }

  return (
    <section className="FieldSet FieldSet--ShippingMethod">
      <div className="FieldSet__Header">
        <div className="FieldSet__Heading">Shipping method</div>
      </div>
      {
        shippingLinesFetching ? <LoadingState />
          : (
            <div className="FieldSet__Content">
              {shippingLines && shippingLines.map((method, index) => (
                <div className="RadioButton" key={index}>
                  <RadioField
                    label={method.description}
                    name="shipping-method"
                    checked={selectedShippingLineIndex === parseInt(method.id, 10)}
                    className="RadioField"
                    disabled={shippingLinesLoadingStatus === 'setting'}
                    onChange={() => setSelectedShippingLine(index)}
                  />
                  <Price className="ShippingMethod__Price" amount={method.amount} />
                </div>
              ))}
            </div>
          )
      }
    </section>
  );
};

ShippingLines.propTypes = {
  showShippingLines: PropTypes.bool,
  shippingLinesFetching: PropTypes.bool,
  shippingLinesLoadingStatus: PropTypes.string,
  shippingLines: PropTypes.array,
  selectedShippingLineIndex: PropTypes.number,
  setSelectedShippingLine: PropTypes.func,
};

const MemoizedShippingLines = React.memo(ShippingLines);

const ShippingLinesContainer = () => {
  const {
    showShippingLines, shippingLinesFetching, shippingLinesLoadingStatus, shippingLines, selectedShippingLineIndex, setSelectedShippingLine,
  } = useShippingLines();

  return (
    <MemoizedShippingLines
      showShippingLines={showShippingLines}
      shippingLinesFetching={shippingLinesFetching}
      shippingLinesLoadingStatus={shippingLinesLoadingStatus}
      shippingLines={shippingLines}
      selectedShippingLineIndex={selectedShippingLineIndex}
      setSelectedShippingLine={setSelectedShippingLine}
    />
  );
};

export default ShippingLinesContainer;
