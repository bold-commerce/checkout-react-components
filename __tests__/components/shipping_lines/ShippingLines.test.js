import React from 'react';
import { render } from '@testing-library/react';
import { ShippingLines } from '../../../src/components/shipping_lines/ShippingLines';
import { exampleShippingLines } from '../../utils/shippingLinesHelper';

describe('ShippingLines', () => {
  test('renders ShippingLines component when shipping lines should be hidden', () => {
    const { asFragment } = render(
      <ShippingLines
        showShippingLines={false}
        shippingLinesFetching={false}
        shippingLinesLoadingStatus=""
        shippingLines={[]}
        selectedShippingLineIndex={0}
        setSelectedShippingLine={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders ShippingLines component', () => {
    const { asFragment } = render(
      <ShippingLines
        showShippingLines
        shippingLinesFetching={false}
        shippingLinesLoadingStatus=""
        shippingLines={exampleShippingLines}
        selectedShippingLineIndex={0}
        setSelectedShippingLine={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders ShippingLines component while shipping lines are being set', () => {
    const { asFragment } = render(
      <ShippingLines
        showShippingLines
        shippingLinesFetching={false}
        shippingLinesLoadingStatus="setting"
        shippingLines={exampleShippingLines}
        selectedShippingLineIndex={0}
        setSelectedShippingLine={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders ShippingLines component while shipping lines are fetching', () => {
    const { asFragment } = render(
      <ShippingLines
        showShippingLines
        shippingLinesFetching
        shippingLinesLoadingStatus=""
        shippingLines={exampleShippingLines}
        selectedShippingLineIndex={0}
        setSelectedShippingLine={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
