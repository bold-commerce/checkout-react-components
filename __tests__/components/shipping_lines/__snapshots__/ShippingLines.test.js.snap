// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingLines renders ShippingLines component 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--ShippingMethod"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Shipping method
      </div>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div
        class="RadioButton"
      >
        <div
          class="stx-field stx-field--with-radio-field RadioField"
        >
          <div
            class="stx-radio-field"
          >
            <span
              class="stx-radio stx-radio--is-checked stx-radio-field__radio"
            >
              <input
                aria-checked="true"
                aria-disabled="false"
                checked=""
                class="stx-radio__input"
                id="stxField1"
                name="shipping-method"
                role="radio"
                type="radio"
                value=""
              />
              <span
                class="stx-radio__box"
              />
            </span>
            <label
              class="stx-radio-field__label"
              for="stxField1"
            >
              Standard Shipping
            </label>
          </div>
        </div>
        <div
          class="stx-price ShippingMethod__Price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            CA$20.00
          </p>
        </div>
      </div>
      <div
        class="RadioButton"
      >
        <div
          class="stx-field stx-field--with-radio-field RadioField"
        >
          <div
            class="stx-radio-field"
          >
            <span
              class="stx-radio stx-radio-field__radio"
            >
              <input
                aria-checked="false"
                aria-disabled="false"
                class="stx-radio__input"
                id="stxField2"
                name="shipping-method"
                role="radio"
                type="radio"
                value=""
              />
              <span
                class="stx-radio__box"
              />
            </span>
            <label
              class="stx-radio-field__label"
              for="stxField2"
            >
              Expedited Shipping
            </label>
          </div>
        </div>
        <div
          class="stx-price ShippingMethod__Price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            CA$40.00
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ShippingLines renders ShippingLines component when shipping lines should be hidden 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--ShippingMethod"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Shipping method
      </div>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div
        class="FieldSet--EmptyState"
      >
        <svg
          fill="none"
          height="29"
          viewBox="0 0 28 29"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#C4C4C4"
            height="5"
            rx="2.5"
            width="28"
          />
          <rect
            fill="#C4C4C4"
            height="5"
            rx="2.5"
            width="28"
            y="8"
          />
          <rect
            fill="#C4C4C4"
            height="5"
            rx="2.5"
            width="28"
            y="16"
          />
          <rect
            fill="#C4C4C4"
            height="5"
            rx="2.5"
            width="28"
            y="24"
          />
          <circle
            cx="2.5"
            cy="10.5"
            fill="white"
            r="1.5"
          />
          <circle
            cx="2.5"
            cy="2.5"
            fill="white"
            r="1.5"
          />
          <circle
            cx="2.5"
            cy="18.5"
            fill="white"
            r="1.5"
          />
          <circle
            cx="2.5"
            cy="26.5"
            fill="white"
            r="1.5"
          />
        </svg>
        To view shipping options, complete filling in your address
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ShippingLines renders ShippingLines component while shipping lines are being set 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--ShippingMethod"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Shipping method
      </div>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div
        class="RadioButton"
      >
        <div
          class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioField"
        >
          <div
            class="stx-radio-field"
          >
            <span
              class="stx-radio stx-radio--is-disabled stx-radio--is-checked stx-radio-field__radio"
            >
              <input
                aria-checked="true"
                aria-disabled="true"
                checked=""
                class="stx-radio__input"
                disabled=""
                id="stxField3"
                name="shipping-method"
                role="radio"
                type="radio"
                value=""
              />
              <span
                class="stx-radio__box"
              />
            </span>
            <label
              class="stx-radio-field__label"
              for="stxField3"
            >
              Standard Shipping
            </label>
          </div>
        </div>
        <div
          class="stx-price ShippingMethod__Price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            CA$20.00
          </p>
        </div>
      </div>
      <div
        class="RadioButton"
      >
        <div
          class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioField"
        >
          <div
            class="stx-radio-field"
          >
            <span
              class="stx-radio stx-radio--is-disabled stx-radio-field__radio"
            >
              <input
                aria-checked="false"
                aria-disabled="true"
                class="stx-radio__input"
                disabled=""
                id="stxField4"
                name="shipping-method"
                role="radio"
                type="radio"
                value=""
              />
              <span
                class="stx-radio__box"
              />
            </span>
            <label
              class="stx-radio-field__label"
              for="stxField4"
            >
              Expedited Shipping
            </label>
          </div>
        </div>
        <div
          class="stx-price ShippingMethod__Price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            CA$40.00
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ShippingLines renders ShippingLines component while shipping lines are fetching 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--ShippingMethod"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Shipping method
      </div>
    </div>
    <div
      class="FieldSet--LoadingState"
    >
      <span
        class="stx-loading-spinner"
      />
    </div>
  </section>
</DocumentFragment>
`;
